Sources/Rendering/Core/Renderer/index.js

```diff
@@ -181,9 +181,13 @@ function vtkRenderer(publicAPI, model) {
     publicAPI.modified();
   };
 
+  publicAPI.hasLight = (light) =>
+    !!model.lights.filter((item) => item === light).length;
   publicAPI.addLight = (light) => {
-    model.lights = [].concat(model.lights, light);
-    publicAPI.modified();
+    if (light && !publicAPI.hasLight(light)) {
+      model.lights = model.lights.concat(light);
```

Comment:
`model.lights.push(light)` is more succinct.

Reply:
Done.